File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        HashSet<String> hashSet = new HashSet<>();
        int cnt = 0;
        while (N-- > 0) {
           String p = br.readLine();
           if(p.equals("ENTER")) {
               hashSet = new HashSet<>();
               continue;
           }
           if(!hashSet.contains(p)) {
               hashSet.add(p);
               cnt++;
           }
       }
        System.out.println(cnt);
    }
}

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
def enter(n):
    s = set()
    cnt = 0
    for i in range(n):
        record = input()
        if record == 'ENTER':
            s.clear() # 초기화  
       	elif record != 'ENTER':
            if record not in s:  
                cnt += 1           
                s.add(record)  
    return cnt

n = int(input())
print(enter(n)) 